src/parameterizations/lateral/MOM_hor_visc.F90

```diff
@@ -2542,54 +2578,54 @@ subroutine smooth_GME(G, GME_flux_h, GME_flux_q)
   real, dimension(SZIB_(G),SZJB_(G)) :: GME_flux_q_original
   real :: wc, ww, we, wn, ws ! averaging weights for smoothing
   integer :: i, j, k, s
-  do s=1,1
-    ! Update halos
+  integer :: is, ie, js, je, Isq, Ieq, Jsq, Jeq
+
+  is  = G%isc  ; ie  = G%iec  ; js  = G%jsc  ; je  = G%jec
+  Isq = G%IscB ; Ieq = G%IecB ; Jsq = G%JscB ; Jeq = G%JecB
+  do s=1,CS%num_smooth_gme
     if (present(GME_flux_h)) then
-      !### Work on a wider halo to eliminate this blocking send!
-      call pass_var(GME_flux_h, G%Domain)
+      ! Update halos if needed
+      if (s >= 2) call pass_var(GME_flux_h, G%Domain)
       GME_flux_h_original(:,:) = GME_flux_h(:,:)
       ! apply smoothing on GME
-      do j = G%jsc, G%jec
-        do i = G%isc, G%iec
+      do j = Jsq, Jeq+1
+        do i = Isq, Ieq+1
           ! skip land points
           if (G%mask2dT(i,j)==0.) cycle
           ! compute weights
           ww = 0.125 * G%mask2dT(i-1,j)
           we = 0.125 * G%mask2dT(i+1,j)
           ws = 0.125 * G%mask2dT(i,j-1)
           wn = 0.125 * G%mask2dT(i,j+1)
-          wc = 1.0 - (ww+we+wn+ws)
-          !### Add parentheses to make this rotationally invariant.
+          wc = 1.0 - ((ww+we)+(wn+ws))
           GME_flux_h(i,j) =  wc * GME_flux_h_original(i,j)   &
-                           + ww * GME_flux_h_original(i-1,j) &
-                           + we * GME_flux_h_original(i+1,j) &
-                           + ws * GME_flux_h_original(i,j-1) &
-                           + wn * GME_flux_h_original(i,j+1)
+                           + ((ww * GME_flux_h_original(i-1,j) &
+                           + we * GME_flux_h_original(i+1,j)) &
+                           + (ws * GME_flux_h_original(i,j-1) &
+                           + wn * GME_flux_h_original(i,j+1)))
         enddo
       enddo
     endif
-    ! Update halos
     if (present(GME_flux_q)) then
-      !### Work on a wider halo to eliminate this blocking send!
-      call pass_var(GME_flux_q, G%Domain, position=CORNER, complete=.true.)
+      ! Update halos if needed
+      if (s >= 2) call pass_var(GME_flux_q, G%Domain, position=CORNER, complete=.true.)
       GME_flux_q_original(:,:) = GME_flux_q(:,:)
       ! apply smoothing on GME
-      do J = G%JscB, G%JecB
-        do I = G%IscB, G%IecB
+      do J = Jsq-1, Jeq
```

Comment:
These q-point loop bounds should probably have symmetric halos around the q-point computational domain extents.